<a href="https://colab.research.google.com/github/nina-mir/scale-hackathon-april-20/blob/main/data/scale_april_17_data_cleaning_filmed_sf.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


mount the drive


In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:

# importing pandas as pd
import pandas as pd
# read_csv function which is used to read the required CSV file
data = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/SF Film Project/scale-hackathon/raw_Film_Locations_in_San_Francisco_20240417.csv')

In [ ]:
print("there are {} columns total.\n\n the columns are:\n\n {} ".format(len(data.columns.values), data.columns.values))

In [ ]:
# data.describe(include='all')
print(F'There are {data.shape[0]} rows and {data.shape[1]} columns in this dataset!')
print(F'The data types of cols are: \n\n {data.dtypes}')
# data.shape

There are 2084 rows and 14 columns in this dataset!
The data types of cols are:

```
Title                            object
Release Year                      int64
Locations                        object
Fun Facts                        object
Production Company               object
Distributor                      object
Director                         object
Writer                           object
Actor 1                          object
Actor 2                          object
Actor 3                          object
SF Find Neighborhoods           float64
Analysis Neighborhoods          float64
Current Supervisor Districts    float64
dtype: object
```

In [ ]:
data[data["Actor 1"]=='Charles Chaplin']

Title  Release Year                Locations  \
927   A Jitney Elopement          1915  20th and Folsom Streets   
1351  A Jitney Elopement          1915         Golden Gate Park   

                                              Fun Facts  \
927                                                 NaN   
1351  During San Francisco's Gold Rush era, the Park...   

                          Production Company           Distributor  \
927   The Essanay Film Manufacturing Company  General Film Company   
1351  The Essanay Film Manufacturing Company  General Film Company   

             Director           Writer          Actor 1         Actor 2  \
927   Charles Chaplin  Charles Chaplin  Charles Chaplin  Edna Purviance   
1351  Charles Chaplin  Charles Chaplin  Charles Chaplin  Edna Purviance   

     Actor 3  SF Find Neighborhoods  Analysis Neighborhoods  \
927      NaN                   53.0                    20.0   
1351     NaN                    9.0                    12.0   

      Current Supervisor Districts  
927                            2.0  
1351                           4.0

In [ ]:
unique = data.describe(include='all').loc['unique']
print(F"The number of unique values in each column are: \n {unique}")

The number of unique values in each column are: 
 Title                            336
Release Year                     NaN
Locations                       1634
Fun Facts                        187
Production Company               208
Distributor                      167
Director                         276
Writer                           294
Actor 1                          255
Actor 2                          263
Actor 3                          156
SF Find Neighborhoods            NaN
Analysis Neighborhoods           NaN
Current Supervisor Districts     NaN
Name: unique, dtype: object


In [ ]:
year_list = data["Release Year"].unique()
year_list_int = [int(i) for i in year_list]
year_list_int.sort()
print(F'the oldest permit was issued in {min(year_list_int)} and the latest permit was issued in {max(year_list_int)}')

the oldest permit was issued in 1915 and the latest permit was issued in 2023


the oldest permit was issued in 1915 and the latest permit was issued in 2023



The total number of unique titles [films/tv series] in this dataset is 336.


In [ ]:
# print(data.info())
# with pd.option_context('float_format', '{:.0f}'.format): print(data.describe())
# print(data.describe())

In [ ]:
target_columns = ['Title', 'Release Year', 'Locations', 'Fun Facts','Director', 'Writer', 'Actor 1', 'Actor 2', 'Actor 3']
# print('data.describe():   ', data[['Title', 'Release Year','Locations', 'Fun Facts','Director', 'Writer', 'Actor 1', 'Actor 2', 'Actor 3']].describe().values)

# print('data.describe():   ', data[['Locations', 'Fun Facts','Director', 'Writer', 'Actor 1', 'Actor 2', 'Actor 3']].describe())

data.describe(include='all').loc['Title']


# describe_list = data[['Locations', 'Fun Facts','Director', 'Writer', 'Actor 1', 'Actor 2', 'Actor 3']].describe().values
# type(data[['Locations', 'Fun Facts','Director', 'Writer', 'Actor 1', 'Actor 2', 'Actor 3']].describe().values.tolist())
# print(describe_list)

In [ ]:
# print('data.info()', data.info())
# data.shape
# data.size
data.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2084 entries, 0 to 2083
Data columns (total 14 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Title                         2084 non-null   object 
 1   Release Year                  2084 non-null   int64  
 2   Locations                     2030 non-null   object 
 3   Fun Facts                     464 non-null    object 
 4   Production Company            2082 non-null   object 
 5   Distributor                   1945 non-null   object 
 6   Director                      2079 non-null   object 
 7   Writer                        2074 non-null   object 
 8   Actor 1                       2080 non-null   object 
 9   Actor 2                       1991 non-null   object 
 10  Actor 3                       1612 non-null   object 
 11  SF Find Neighborhoods         1963 non-null   float64
 12  Analysis Neighborhoods        1963 non-null   float64
 13  Cur

Let's abridge this data set by getting rid of the following columns at this moment:
```
'SF Find Neighborhoods',
'Analysis Neighborhoods',
'Current Supervisor Districts',
'Production Company',
'Distributor',
'Actor 2',
'Actor 3',
'Writer',
'Fun Facts'
```

The resulting ```abridged_data``` will be of only 5 columns!

In [ ]:
abridged_data = data.drop(columns=['SF Find Neighborhoods','Analysis Neighborhoods',
                                   'Current Supervisor Districts', 'Production Company',
                                   'Distributor', 'Actor 2', 'Actor 3', 'Writer', 'Fun Facts'])

In [ ]:
abridged_data.info(verbose="true")

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2084 entries, 0 to 2083
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Title         2084 non-null   object
 1   Release Year  2084 non-null   int64 
 2   Locations     2030 non-null   object
 3   Director      2079 non-null   object
 4   Actor 1       2080 non-null   object
dtypes: int64(1), object(4)
memory usage: 81.5+ KB


let's drop any rows that has any mepty values

Get rid of NaN values by replacing them with empty space


In [ ]:
abridged_data.dropna(axis=0, inplace=True)
abridged_data.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
Index: 2021 entries, 0 to 2083
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Title         2021 non-null   object
 1   Release Year  2021 non-null   int64 
 2   Locations     2021 non-null   object
 3   Director      2021 non-null   object
 4   Actor 1       2021 non-null   object
dtypes: int64(1), object(4)
memory usage: 94.7+ KB


In [ ]:
abridged_data.isna().sum()

Title           0
Release Year    0
Locations       0
Director        0
Actor 1         0
dtype: int64

In [ ]:
abridged_data.sort_values(by=["Release Year"], inplace=True)

In [ ]:
abridged_data.head()

Title  Release Year                              Locations  \
927   A Jitney Elopement          1915                20th and Folsom Streets   
1351  A Jitney Elopement          1915                       Golden Gate Park   
1464               Greed          1924                 Hayes Street at Laguna   
266                Greed          1924                Bush and Sutter Streets   
751                Greed          1924  Cliff House (1090 Point Lobos Avenue)   

               Director          Actor 1  
927     Charles Chaplin  Charles Chaplin  
1351    Charles Chaplin  Charles Chaplin  
1464  Eric von Stroheim       Zasu Pitts  
266   Eric von Stroheim       Zasu Pitts  
751   Eric von Stroheim       Zasu Pitts

The abridged dataset has 279 unique titles

In [ ]:
abridged_data.describe(include='all').loc['unique']

Title            279
Release Year     NaN
Locations       1628
Director         233
Actor 1          213
Name: unique, dtype: object

Let's re-set index then

Store this 279 titles to file ...

In [ ]:
abridged_data.reset_index(inplace=True, drop=True)

In [ ]:
abridged_data.to_csv('abridged_april_17_279_Titles.csv')

In [ ]:
slice = abridged_data.loc[:100]
slice.to_csv('slice_abridged_april_17_279_Titles.csv')